lib/lib.go

```diff
@@ -44,6 +44,10 @@ var (
 	// directory. only $HOME is replaced (no other $ env vars).
 	defaultIPFSLocation = "$HOME/.ipfs"
 
+	// ErrNotLinkedToFilesystem is the err implementers should return when we
+	// are expecting the dataset to have a file system link, but fsiPath is empty
+	ErrNotLinkedToFilesystem = fmt.Errorf("dataset is not linked to the filesystem")
```

Comment:
Let's move this into the fsi package so fsi can return it as well. It'll require you to import the FSI package within `api/datatasets.go`, I think that's ok.

fsi/fsi.go:
```go
 // ErrNotLinkedToFilesystem is the err implementers should return when we
 // are expecting the dataset to have a file system link, but fsiPath is empty
 ErrNoLink = fmt.Errorf("dataset is not linked to the filesystem")
```